### PR TITLE
minor code improvement (in fetch credentials() )

### LINKED ISSUE #169: Minor code improvement
Снимок экрана 2019-06-21 в 18 14 56

### PR DESCRIPTION
**Pull request by eugenosm**
closes #169

### CODE REVIEW

File: agent/__init__.py [hunk 1]
```diff
@@ -542,14 +542,16 @@ def clear_credentials(path):
...
 
-            pw = pwd.getpwnam("root")
+            pw = root_pw  # if no owner, use 'root'
             if owner:
                 try:
                     pw = pwd.getpwnam(owner)
-                except KeyError:
-                    pw = pwd.getpwnam("root")
+                except KeyError:  # if invalid owner, use 'root'
+                    pw = root_pw
```
Comment (a-martynovich) [inline agent/__init__.py]: `pass`

Comment (eugenosm) [reply agent/__init__.py]: may be so.
I assumed a slightly different behavior. But I tend to agree with you.

Comment (eugenosm) [reply agent/__init__.py]: changed

Comment (a-martynovich) [reply agent/__init__.py]: So now we just skip a credential if its owner doesn't exist?

Comment (eugenosm) [reply agent/__init__.py]: yes. exactly.

File: agent/__init__.py [hunk 2]
```diff
@@ -542,14 +542,17 @@ def clear_credentials(path):
...
 
-            pw = pwd.getpwnam("root")
+            pw = root_pw  # if no owner, use 'root'
             if owner:
                 try:
                     pw = pwd.getpwnam(owner)
-                except KeyError:
-                    pw = pwd.getpwnam("root")
+                except KeyError:  # if invalid owner, use 'root'
+                    print("Warning. There are credentials with wrong owner ({}). Skipped.".format(owner))
```
Comment (vpetersson) [inline agent/__init__.py]: That comment seems to be inconsistent with what was discussed in the comment.

Comment (eugenosm) [reply agent/__init__.py]: removed

**Pull request merged**